EncoreTickets.SDK/BaseApi.cs

```diff
@@ -150,69 +152,73 @@ public Stream GetResource(string url)
             else
             {
                 rr = client.Execute(request);
-                T rawData = (rr.IsSuccessful) ? SimpleJson.SimpleJson.DeserializeObject<T>(rr.Content) : null;
+                var rawData = (rr.IsSuccessful) ? SimpleJson.SimpleJson.DeserializeObject<T>(rr.Content) : null;
                 apiResponse = new ApiResponse<T>(rawData);
             }
 
-            // IRestResponse respone = client.Execute(request);
-
-            //SimpleJson.JsonArray x = SimpleJson.SimpleJson.DeserializeObject<SimpleJson.JsonArray>(respone.Content);
-            //AvailabilityResponse ar = (AvailabilityResponse)x;
-
-
-            //IRestResponse<ApiResponse<T>> response2 = (ApiResponse<T>)
-
+            return new ApiResultList<T>(context, request, rr, apiResponse);
+        }
 
+        /// <summary>
+        /// Get a rest client
+        /// </summary>
+        /// <returns></returns>
+        private RestClient GetClient()
+        {
+            var rc = new RestClient("https://" + string.Format(host, context.envrionment))
+            {
+                Authenticator = GetAuthenticator()
+            };
+            return rc;
+        }
 
+        private IAuthenticator GetAuthenticator()
+        {
+            if (!string.IsNullOrEmpty(context.accessToken))
+            {
+                return new JwtAuthenticator(context.accessToken);
+            }
 
-            return new ApiResultList<T>(this.context, request, rr, apiResponse);
+            return !string.IsNullOrEmpty(context.userName)
+                ? new HttpBasicAuthenticator(context.userName, context.password)
+                : null;
         }
 
         /// <summary>
         /// Get the request
         /// </summary>
         /// <param name="endpoint"></param>
+        /// <param name="method"></param>
+        /// <param name="body"></param>
         /// <returns></returns>
-        private RestRequest GetRequest(string endpoint)
+        private RestRequest GetRequest(string endpoint, Method method, object body)
         {
-            var request = new RestRequest(endpoint, Method.GET);
-
-            // add HTTP Headers
-            request.AddHeader("x-SDK", "EncoreTickets.SDK.NET v1");  // todo: add build numers
+            var request = new RestRequest(endpoint, method);
+            AddHeadersToRequest(request);
```

Comment:
just a question, in C# do we not use object reference when we invoke a method? if the method is in the class, do we not have `this`? just asking cos i don't have C# experience

Reply:
Yes, `this` can be used, but not necessary, because the class has direct access to its available methods.